### PR TITLE
Added a way to run user code. 

### PR DESCRIPTION
**Pull request by JacobSiegel35**
Still need to figure out where to display output on webpage. Right now it is just being logged to the web console.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -2,16 +2,48 @@ const express = require('express');
...
+const fs = require('fs');
+const { exec } = require('child_process');
+const { FILE } = require('dns');
 
 // serve the react app
 const buildPath = path.join(__dirname, 'build');
 
 app.use(express.static(buildPath));
 
-app.get('/*', (req, res) => {
```
Comment (bsiegel44) [inline server.js:11]: Removing this may break react router. Did you test?

File: server.js [hunk 2]
```diff
@@ -2,16 +2,48 @@ const express = require('express');
...
+  const FILE_NAME = "code" + req.body.roomID + ".py";
+
+  let status = 200;
+
+  fs.writeFile(FILE_NAME, code, { flag: 'w' }, e => {
+    if (e) {
+      status = 500;
+    }
+
+    exec('python ' + FILE_NAME, (err, stdout, stderr) => {
```
Comment (bsiegel44) [inline server.js:31]: is this a good idea?

File: package.json [hunk 3]
```diff
@@ -6,6 +6,7 @@
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^9.5.0",
     "@testing-library/user-event": "^7.2.1",
+    "body-parser": "^1.19.0",
```
Comment (bsiegel44) [inline package.json:9]: express already includes an implementation of body-parser

File: server.js (hunk 2 above)
Comment (JacobSiegel35) [reply server.js:31]: Yes. Who cares if someone tries to bring down a free heroku server?

File: server.js (hunk 1 above)
Comment (JacobSiegel35) [reply server.js:11]: You can change it back if you want. Should still work

File: package.json (hunk 3 above)
Comment (JacobSiegel35) [reply package.json:9]: Are you sure? I looked on Stackoverflow and it looks like body-parser is required to look at bodies of http requests

Comment (bsiegel44) [reply package.json:9]: http://expressjs.com/en/4x/api.html#express.json

**Pull request merged**